packages/webapp/src/pages/VerifyEmail.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { Loading } from '@geist-ui/core';
+import { toast } from 'react-toastify';
+
+import DefaultLayout from '../layout/DefaultLayout';
+
+export default function VerifyEmail() {
+    const [serverErrorMessage, setServerErrorMessage] = useState('');
+    const [email, setEmail] = useState('');
+    const [loaded, setLoaded] = useState(false);
+    const navigate = useNavigate();
+
+    const { uuid } = useParams();
+
+    useEffect(() => {
+        if (!uuid) {
+            navigate('/');
+        }
+    }, [uuid, navigate]);
+
+    useEffect(() => {
+        const getEmail = async () => {
+            const res = await fetch(`/api/v1/account/email/${uuid}`);
+            const response = await res.json();
+
+            if (res?.status === 200) {
+                const { email, verified } = response;
+
+                if (verified) {
+                    setServerErrorMessage('Email already verified. Please navigate to the login page');
+                }
+                setEmail(email);
+            } else {
+                setServerErrorMessage(response.error.message || 'Issue verifying email. Please try again.');
+            }
+            setLoaded(true);
+        };
+
+        if (!loaded) {
+            getEmail();
+        }
+    }, [uuid, loaded, setLoaded]);
+
+    const resendEmail = async (e: React.SyntheticEvent) => {
+        e.preventDefault();
+        setServerErrorMessage('');
+
+        const res = await fetch('/api/v1/account/resend-verification-email', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                uuid
+            })
+        });
+
+        if (res?.status === 200) {
+            toast.success('Verification email sent again!', { position: toast.POSITION.BOTTOM_CENTER });
+        } else {
+            const response = await res.json();
+            setServerErrorMessage(response.error.message || 'Unkown error...');
+        }
+    };
+
+    if (!loaded) {
```

Comment:
this is seems more related to the button that the page, it flickers when clicking

Reply:
Couldn't reproduce. Can you show an example of this?

Reply:
https://github.com/NangoHQ/nango/assets/1637651/f7fc5775-3087-4e14-a2b1-878ee5fc86ec